queries.sql

```diff
@@ -14,3 +15,83 @@ SELECT * FROM animals WHERE neutered =true;
 
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/* -- Transaction update the animals table -- */
+BEGIN;
+UPDATE animals SET species ='unspecified';  --made update on species column
+SELECT species FROM animals; --verify the change was made
+ROLLBACK;
+SELECT species FROM animals; --verify roll back the change
+
+/*-- Update animals table and Commit --*/ 
+
+BEGIN;
+UPDATE animals SET species ='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species ='pokemon' WHERE species IS NULL;
+SELECT name,species from animals; --verify update
+COMMIT;
+SELECT name,species from animals; --verify update made after commit
+
+/*-- Delete all animals data inside transaction and undo deletion --*/
+BEGIN;
+DELETE FROM animals;
+SELECT COUNT(*) FROM animals; --check the count of animals 
+SELECT * FROM animals; -- check after deletion
+ROLLBACK;
+
+/*-- Check the count of animals after Rollback --*/
+SELECT COUNT(*) FROM animals; 
+
+/*-- Delete all animals data inside transaction and undo deletion end --*/
+
+/*-- Delete all animals born after jan 1st,2022 --*/
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth >='01/01/2022';
+SELECT name, date_of_birth FROM animals;
+
+--create savepoint 
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg =weight_kg * -1;
+SELECT name, weight_kg FROM animals;
+ ROLLBACK TO sp1;
+ SELECT name, weight_kg FROM animals; -- Check the result after rollback
+
+ UPDATE animals SET weight_kg =weight_kg * -1
+   WHERE weight_kg < 0;
+ SELECT name, weight_kg 
+   FROM animals;  --Check the change made
+ COMMIT;
+
+/*-- Aggreigate --*/
+ /*-- How many animals are there --*/
+ 
+ SELECT COUNT(*) from animals;
+ 
+ /*-- Count animals never tried to escape --*/
+ 
+  SELECT COUNT(*) from animals WHERE escape_attempts =0;
+  
+  /*-- Calculate the average weight of all animals --*/
+  
+  SELECT AVG(weight_kg) from animals;
+  
+ /*-- Find the max escape attemps ,neutered or not neutered animals --*/
+ 
+ SELECT neutered, 
+  MAX(escape_attempts) AS max_escap_attempts
+   FROM animals 
+   GROUP BY neutered;
+   
+/*-- Find the max weight of animals --*/
+ 
+SELECT MAX(weight_kg) FROM animals;
+
+/*-- Find the min weight of animals --*/
+
+SELECT MIN(weight_kg) FROM animals;
```

Comment:
- [ ] Kindly do those two query in one query and right implement it as they asked in this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=What%20is%20the%20minimum%20and%20maximum%20weight%20of%20each%20type%20of%20animal%3F) to get the minimum and maximum of each type of animal, not the minimum and maximum of animals table. 
### Hint: you need to use the `GROUP BY` statement.